packages/solana/src/connectors/BaseConnector.ts

```diff
@@ -0,0 +1,392 @@
+      transaction.feePayer = transferParams.feePayer === 'from' ? fromPubkey : toPubkey
+    } else if (type === 'program') {
+      const programParams = params as TransactionArgs['program']['params']
+      transaction.add(
+        new TransactionInstruction({
+          keys: [
+            { pubkey: fromPubkey, isSigner: true, isWritable: programParams.isWritableSender }
+          ],
+          programId: new PublicKey(programParams.programId),
+          data: Buffer.from(base58.decode(JSON.stringify(programParams.data)))
+        })
+      )
+      transaction.feePayer = fromPubkey
+    } else {
+      throw new Error(`No transaction configuration for type ${String(type)}`)
+    }
+
+    const response = await this.requestCluster('getLatestBlockhash', [{}])
+
+    const { blockhash: recentBlockhash } = response.value
+    transaction.recentBlockhash = recentBlockhash
+
+    return transaction
+  }
+
+  protected async constructVersionedTransaction(params: TransactionArgs['transfer']['params']) {
+    const fromAddress = SolStoreUtil.state.address
+    if (!fromAddress) {
+      throw new Error('No address connected')
+    }
+    const fromPubkey = new PublicKey(fromAddress)
+    const toPubkey = new PublicKey(params.to)
+
+    const instructions = [
+      SystemProgram.transfer({
+        fromPubkey,
+        toPubkey,
+        lamports: params.amountInLamports
+      })
+    ]
+
+    const response = await this.requestCluster('getLatestBlockhash', [{}])
+    const { blockhash: recentBlockhash } = response.value
+
+    const messageV0 = new TransactionMessage({
+      payerKey: fromPubkey,
+      recentBlockhash,
+      instructions
+    }).compileToV0Message()
+
+    // Make a versioned transaction
+    const transactionV0 = new VersionedTransaction(messageV0)
+
+    return transactionV0
+  }
+
+  /*
+   * Public async sendTransaction(encodedTransaction: string) {
+   * const signature = await this.requestCluster('sendTransaction', [encodedTransaction])
+   * return signature
+   * }
+   */
+
+  public async getTransaction(transactionSignature: string) {
+    const transaction = await this.requestCluster('getTransaction', [
+      transactionSignature,
+      { encoding: 'jsonParsed', commitment: 'confirmed' }
+    ])
+
+    return transaction
+  }
+
+  public async watchTransaction(
+    transactionSignature: string,
+    callback: (params: Transaction | number) => void
+  ) {
+    return this.subscribeToCluster('signatureSubscribe', [transactionSignature], callback)
+  }
+
+  public async getBalance(requestedAddress: string, currency: 'lamports' | 'sol' = 'sol') {
+    try {
+      const address = requestedAddress ?? SolStoreUtil.state.address
+      const balance = await this.requestCluster('getBalance', [
+        address,
+        { commitment: 'processed' }
+      ])
+      const formatted =
+        currency === 'lamports'
+          ? `${balance?.value || 0} lamports`
+          : `${(balance?.value || 0) / 1000000000} sol`
```

Comment:
Why this is `1000000000` and would it be better to define this variable on the top with a meaningful name?

```
// or something else name
const BALANCE_VALUE_DECIMAL_DIVIDER = 1000000000
```